### ISSUE TITLE
[arm32/Ubuntu] R2R test failures

### ISSUE THREAD

**Issue by BruceForstall** [dotnet/coreclr#16965]
A basic (non-stress) R2R test run on arm32/Ubuntu shows 198 test failures:

https://ci.dot.net/job/dotnet_coreclr/job/dev_unix_test_workflow/job/arm_cross_checked_ubuntu_r2r_tst_prtest/1/

Stress modes show similar, but slightly different, failure numbers.

There are several failure modes in addition to simple segmentation fault. One common mode is:
```
               Assert failure(PID 3028 [0x00000bd4], Thread: 3028 [0x0bd4]): offset >= - MAX_OFFSET_UNCONDITIONAL_BRANCH_THUMB && offset < MAX_OFFSET_UNCONDITIONAL_BRANCH_THUMB
                   File: /mnt/j/workspace/dotnet_coreclr/dev_unix_test_workflow/arm_cross_checked_ubuntu_prtest/src/vm/arm/cgencpu.h Line: 285
                   Image: /ssd/j/workspace/dotnet_coreclr/dev_unix_test_workflow/arm_cross_checked_ubuntu_r2r_tst_prtest/bin/tests/Linux.arm.Checked/Tests/Core_Root/corerun
```

Another:
```
               Assert failure(PID 2726 [0x00000aa6], Thread: 2726 [0x0aa6]): Reached the "unreachable": FAILED: <unreachable>
                   File: /mnt/j/workspace/dotnet_coreclr/dev_unix_test_workflow/arm_cross_checked_ubuntu_prtest/src/vm/exceptionhandling.cpp Line: 4620
                   Image: /ssd/j/workspace/dotnet_coreclr/dev_unix_test_workflow/arm_cross_checked_ubuntu_r2r_tst_prtest/bin/tests/Linux.arm.Checked/Tests/Core_Root/corerun
```

Another:
```
               Assert failure(PID 29927 [0x000074e7], Thread: 29927 [0x74e7]): pImportSection == pModule->GetImportSectionForRVA(rva)
                   File: /mnt/j/workspace/dotnet_coreclr/dev_unix_test_workflow/arm_cross_checked_ubuntu_prtest/src/vm/prestub.cpp Line: 2128
                   Image: /ssd/j/workspace/dotnet_coreclr/dev_unix_test_workflow/arm_cross_checked_ubuntu_r2r_tst_prtest/bin/tests/Linux.arm.Checked/Tests/Core_Root/corerun
```

**Comment by RussKeldorph** [comment]
@janvorli @jkotas Does it make sense to start with `area-VM` unless/until proven otherwise?

**Comment by jkotas** [comment]
`area-ReadyToRun` ?

**Comment by RussKeldorph** [comment]
@jkotas Thanks...forgot about that one.

**Comment by sandreenko** [comment]
I do not see any hits of `assert(pImportSection == pModule->GetImportSectionForRVA(rva))` in the last run [output](https://ci.dot.net/job/dotnet_coreclr/job/dev_unix_test_workflow/job/arm_cross_checked_ubuntu_r2r_tst_prtest/3/consoleText). Maybe the issue run did not include #16171 fix.

**Comment by BruceForstall** [comment]
I kicked off some CI runs after the latest changes:

https://ci.dot.net/job/dotnet_coreclr/job/master/job/arm_cross_checked_ubuntu_r2r_flow/
https://ci.dot.net/job/dotnet_coreclr/job/master/job/jitstress/job/arm_cross_checked_ubuntu_r2r_jitminopts_flow/
https://ci.dot.net/job/dotnet_coreclr/job/master/job/jitstress/job/arm_cross_checked_ubuntu_r2r_jitstress1_flow/
https://ci.dot.net/job/dotnet_coreclr/job/master/job/jitstress/job/arm_cross_checked_ubuntu_r2r_jitstress2_flow/

**Comment by BruceForstall** [comment]
These show about 50 errors, but the JIT stress jobs timed out.

**Comment by janvorli** [comment]
The remaining errors were fixed by #17292
